### PR TITLE
#1900 Refactor StreamCollie to remove redundant code

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #1900)
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
Refactor only, no new implementation
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-agent/src/main/scala/com/island/ohara/agent/StreamCollie.scala [hunk 1]
```diff
@@ -81,6 +182,36 @@ trait StreamCollie extends Collie[StreamClusterInfo, StreamCollie.ClusterCreator
...
+    */
+  protected def serviceName: String
+
+  protected def doCreator(executionContext: ExecutionContext,
+                          clusterName: String,
+                          containerName: String,
+                          containerInfo: ContainerInfo,
+                          node: Node,
+                          jmxPort: Int,
+                          route: Map[String, String]): Unit
```
Comment (chia7712) [inline ohara-agent/src/main/scala/com/island/ohara/agent/StreamCollie.scala]: consider using Future[Unit] to replace Unit as our production code (k8s) is based on async function.

Comment (saivirtue) [reply ohara-agent/src/main/scala/com/island/ohara/agent/StreamCollie.scala]: Sure. I will modified other collie also

**Comment by saivirtue** [comment]
retry

**Comment by saivirtue** [comment]
the failed test is traced on #2072, not related to this PR.
merge this PR later.

**Pull request merged**